Giant Spider Variants + Venom modification

Shaddap1 opened the pull request:
## About the PR
Adds 4 new spiders to the game using the existing tg sprites in the repo
- Hunter, Guard, Midwife, and Nurse

Adds these new spiders to the SpiderSpawn gamerule, each having a probability of 0.01

Tweaks the way spiders inject their venom.  
- Before, all spiders have infinite venom and injected a small amount of venom with every bite.  
- Now, venom pools regenerate slowly over time and spiders inject most if not all of their venom in one bite.  
- Mechanotoxin damage has been reduced to maintain prior balance.  1u -> 0.6 poison @ 0.2u/s
- Mechanotoxin now clots blood, preventing bleeding and allowing venom to stay in the bloodstream instead of removed to blood puddles.

Gives each variant of giant spider different attributes:
- Tarantula: Average, weak venom. Piercing reduced from 6 to 4, 1 weak venomous bite every 50 seconds.
- Hunter Spider: Fast, aggressive, and fragile.  Movespeed increased from 4 to 6, health reduced from 90 to 60, pierce increased from 6 to 8, 1 strong venomous bite every 100 seconds.
- Guard Spider: Slow, defensive, and tanky.  Movespeed reduced from 4 to 3, health increased from 90 to 180.  pierce reduced from 6 to 4, 1 weak venomous bite every 5 seconds
- Midwife Spider: Venomous and defensive.  Health increased from 90 to 120, pierce reduced from 6 to 4, 6 weak venomous bites every 150 seconds
- Nurse Spider: Venomous and fragile:  Speed increased from 4 to 5, health reduced from 90 to 60, infinite very weak venomous bites.

Clown spider changes:
- Venom changed from mechanotoxin to laughter.  Victims now laugh uncontrollably.

## Why / Balance
Giant Spiders are the among most common hostile animal threat encountered by crew.  Currently we have two variants: tarantulas and clown spiders.  This PR refreshes those encounters with new sprites and minor gameplay variation.  Hunter spiders and Midwife spiders will ideally be high-priority callouts and incentivize urgent responses from security, whereas the guard spider is more incentivized to take over a small corner of maints.  Medical departments will have minor variations in the kinds of spider bite victims they will have to treat.  Some crew will have more serious cases of envenoming than others, enhancing triage pressure.  I also think the unused sprites from tg just look neat.

Spiders are generally weaker in combat with this PR, but as they inject more venom, players have to become aware of the consequences of staying to fight versus running away and seeking medical treatment quickly.  Some spiders are more of an immediate threat than others.  The nurse and guard spiders will have a harder time sowing chaos whereas the hunter spider is actually pretty terrifying. 

If this PR is merged and well received by the community, I will look at giving each spider their own kind of venom, which will be hell to microbalanace.


## Technical details

- Moved spider definitions from animals.yml to new spiders.yml for better organization.
- Added new spider definitions
- Changed clown spider ID from MobClownSpider to MobGiantSpiderClown to match spider naming conventions
- Removed SolutionContainerManager, and other venom-related components from MobSpiderBase for non-venomous spiders.
- Adjusted references to old MobClownSpider ID to use new MobGiantSpiderClown
- Removed default "angry" inheritance from MobGiantSpiderClown and added MobGiantSpiderClownAngry for SpiderClownSpawn gamerule to match conventions for other spiders.

## Media
Nurse Spider
<img width="64" height="64" alt="nurse" src="https://github.com/user-attachments/assets/85d63152-81f0-44f3-8b47-3f1f7033e3fe" />
Midwife Spider
<img width="64" height="64" alt="midwife" src="https://github.com/user-attachments/assets/fd4926c2-973e-4932-8b53-1a66beafbb99" />
Guard Spider
<img width="64" height="64" alt="guard" src="https://github.com/user-attachments/assets/95662577-d75c-4dec-803c-38f20fc4281d" />
Hunter Spider
<img width="64" height="64" alt="hunter" src="https://github.com/user-attachments/assets/1528425a-be19-4c00-8a8e-685666305073" />

Testing SpiderSpawn gamerule
![spiderspawnevent](https://github.com/user-attachments/assets/6e165350-04b7-4b6f-93bf-31ec36dd05b5)


## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.


## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. --> Clown Spider ID changed from MobClownSpider to MobGiantSpiderClown



**Changelog**
:cl:
- add: Four new spider variants that can appear during spider infestations: Hunter, Guard, Midwife, and Nurse.
- tweak: Spiders now inject much more venom, but their venom regenerates more slowly over time
- tweak: Mechanotoxin damage reduced from 1.6 to 0.6
- tweak: Mechanotoxin now thickens blood and reduces bleeding
- tweak: Spider Clowns now inject Laughter instead of Mechanotoxin, causing victims to giggle uncontrollably.

Commit:
move spider entity definitions to spiders.yml

Commit:
Spider Variation

Commit:
Lower hunter venom regeneration and fix clown spider spawner reference

EthanQix commented:
> imo they should have different abilities based on their name, might not be for this pr tho, idk could also be a thing like a swarm antag for the crew to fight, where theres a spider queen the crew has to find and kill. idk

You mean like the spider swarm antag from SS13 ? ^^

I really like the detail of the mechanotoxin thickening blood, but I wonder if the poison damage numbers will need some balance now that it won't compound with bloodloss damage.

Shaddap1 commented:
> imo they should have different abilities based on their name, might not be for this pr tho, idk could also be a thing like a swarm antag for the crew to fight, where theres a spider queen the crew has to find and kill. idk

I looked at giving the guard spider a reduced cooldown for placing webs but I didn't want to touch actions as they are currently undergoing a refactor.

I've also submitted a design doc for spider queen. 
https://github.com/space-wizards/docs/pull/533

Nukesthestation commented:
> You mean like the spider swarm antag from SS13 ? ^^

thats a thing?

Shaddap1 commented:
> > imo they should have different abilities based on their name, might not be for this pr tho, idk could also be a thing like a swarm antag for the crew to fight, where theres a spider queen the crew has to find and kill. idk
> 
> You mean like the spider swarm antag from SS13 ? ^^
> 
> I really like the detail of the mechanotoxin thickening blood, but I wonder if the poison damage numbers will need some balance now that it won't compound with bloodloss damage.

From my testing with the default spiders, average poison damage per bite would ramp down from 5 poison per bite at no bleed stacks to around 1.5 at full bleed stacks.  

With the exception of the hunter spider, spiders have a lower time-to-crit against crew but because they inject more venom and the venom is metabolized slowly, crew need to seek treatment for the spiderbites.  From my testing against AI spiders, you can kill most spiders 1v1 with a crowbar and end up at poor health, with around 1 minute or more until you reach bad or critical.  I think the balance changes are properly addressed.

Shaddap1 commented:
> Now we just need a way for spiders to heal themselves. Soon my fantasy of playing a spider like a major antagonist will be fully realized...

They do have very weak passive regeneration (4 brute/heat per minute).  I was tempted to lower health across the board and increase regeneration but this PR is already very large.

kin98 commented:
> [X ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
[ X] I have added media to this PR or it does not require an ingame showcase.

could you remove the white space so the checkbox is looking like it should

- [X] so
- [ X] not so

Princess-Cheeseballs commented on Resources/Prototypes/Entities/Mobs/NPCs/spiders.yml:
```diff
@@ -0,0 +1,542 @@
+  - type: MeleeChemicalInjector
+    transferAmount: 10
+    solution: melee
+  - type: MovementSpeedModifier
```
This gets overwritten by bodysystem and wont do anything. 
There's a couple ways around this.

1. New Body prototypes for every spider (yuck)
2. New Body prototype which doesn't use limbs to calculate speed (jank)
3. Comment this out and cry. 

Pick your poison.

Princess-Cheeseballs replied:
Ah, their required legs are already set to 0 so yeah nevermind Option 2 was chosen for you.

Princess-Cheeseballs reviewed (changes requested):
Some stuff, also your tests are failing please fix them.

Shaddap1 committed:
Update toxins.yml

Adjusted movement speed modifier condition from min 9.4 to 9.5 per requested change

Shaddap1 commented:
> > > Someday, sometime [space-wizards/docs#72](https://github.com/space-wizards/docs/pull/72) will be merged
> > 
> > 
> > See [space-wizards/docs#533](https://github.com/space-wizards/docs/pull/533) which imho is a more comprehensive design doc.
> 
> They're simply completely different, and both have a right to exist in the same game. Perhaps, if your design is accepted, it would be worthwhile to give regular spiders some abilities, such as healing for the Nurse Spider and the ability for the Midwife Spider to lay regular spider eggs after, say, cocoon victims.

I think they are functionally very similar.  Midround antags that build a swarm with low snowball potential.  Spider Queen is pretty much analogous to the Broodmother but is intended to have a diverse kit.

I would love to dial up the differences between the spider types more, but actions are currently frozen pending refactor.  The broodlings in the Spider Queen design doc could be easily replaced with the spiders in this PR and they could be shifted closer to the tg version.  I don't think they are different enough to justify having both.

Shaddap1 committed:
Update MinorAntagonists.xml

fixed guidebook reference from GiantSpiderAngry to MobGiantSpiderTarantulaAngry

Golubgik commented:
> 
> I would love to dial up the differences between the spider types more, but actions are currently frozen pending refactor.  The broodlings in the Spider Queen design doc could be easily replaced with the spiders in this PR and they could be shifted closer to the tg version.  I don't think they are different enough to justify having both.

Oh, I would also like to ask, what about spiderlings?

Princess-Cheeseballs commented on Resources/Prototypes/Entities/Mobs/NPCs/spiders.yml:
```diff
@@ -0,0 +1,542 @@
+  parent:
+  - MobSpiderAngry
+  id: MobSpiderAngryBase
+  abstract: true
```
I'd rather not have an abstract entity which just inherits from another abstract entity and nothing else.

Princess-Cheeseballs commented on Resources/Prototypes/Entities/Mobs/NPCs/spiders.yml:
```diff
@@ -0,0 +1,542 @@
+        maxVol: 30
+        reagents:
+          - ReagentId: Mechanotoxin
+            Quantity: 10
```
I'll sit on this one but it might just need me to bite the bullet on proper solution prototypes sooner rather than later.

Shaddap1 committed:
Merge branch 'master' into spider-variation

Princess-Cheeseballs reviewed (approved):
Needs art approval and merge conflicts resolved, but otherwise looks good.
